tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -331,3 +331,18 @@ TEST(SGSparseMatrix, equals_different_shape)
 	EXPECT_FALSE(m1.equals(m2));
 	EXPECT_FALSE(m2.equals(m1));
 }
+
+TEST(SGSparseMatrix, pointe_equal)
+{
+	SGSparseMatrix<float64_t> m1(2, 2);
+	SGSparseMatrix<float64_t> m2(2, 3);
+	EXPECT_FALSE(m1 == m2);
+	EXPECT_TRUE(m1 == m1);
+	EXPECT_TRUE(m2 == m2);
+
+	SGSparseMatrix<float64_t> m3(2, 2);
+	EXPECT_FALSE(m1 == m3);
+
+	SGSparseMatrix<float64_t> m4(m1);
+	EXPECT_TRUE(m1 == m4);
```

Comment:
Although I really like that you though of this case, I think this is a test for the copy constructor and not for ==.
Would remove it

Reply:
Removed. Please take a look.